tactics/hints.ml

```diff
@@ -66,6 +66,9 @@ let decompose_app_bound t =
     | Proj (p, c) -> ConstRef (Projection.constant p), Array.cons c args
     | _ -> raise Bound
 
+let secvars_of_hyps hyps =
+  List.fold_left (fun l id -> Id.Pred.add id l) Id.Pred.empty hyps
+
```

Comment:
I don't understand this function. It collects all vars of hyps w/o filtering on section variables. Is it what it is expected to do?